sound/soc/sof/intel/hda-stream.c

```diff
@@ -462,47 +462,30 @@ irqreturn_t hda_dsp_stream_interrupt(int irq, void *context)
 {
 	struct hdac_bus *bus = context;
 	struct sof_intel_hda_dev *sof_hda = bus_to_sof_hda(bus);
-	u32 stream_mask;
 	u32 status;
-
-	if (!pm_runtime_active(bus->dev))
-		return IRQ_NONE;
+	int ret = IRQ_WAKE_THREAD;
 
 	spin_lock(&bus->reg_lock);
```

Comment:
The locking is also wrong here now. The hard ISR is now taking the lock **only** to read INTSTS, which is already suspicious, but can have valid use cases, I think. But below in the thread you now read INTSTS without taking any locks. That seems wrong to me.